mesonbuild/interpreter.py

```diff
@@ -2589,28 +2540,20 @@ def process_new_values(self, invalues):
             elif isinstance(v, build.Data):
                 self.build.data.append(v)
             elif isinstance(v, dependencies.ExternalProgram):
-                return ExternalProgramHolder(v, self.subproject)
+                pass
```

Comment:
Could this case be combined with the final `elif`?

Reply:
Could be a nice little cleanup indeed.